### DISCUSSION TITLE
concatenate var1 websend

### DISCUSSION THREAD

**Discussion by vodkawasserfall** [arendst/Tasmota#16570] [Q&A] [answered]
is it possible to publish the DHCP IP via http get ?

i've tried several syntax variants found on the internet, smth like this:

`websend[192.10.1.2]/?device=a&ipaddress=%IPAddress1%`

```
Var1 IPAddress1
websend[192.10.1.2]/?device=a&ipaddress=%Var1%
 + Var1
 Var1
(%var1%)
```

**Comment by barbudor** [comment, marked as answer]
You are inventing syntax that do not exist
You need to capture the output of IPAddress 1 output but you need to execute the rule at the right time
So I suggest
```
Rule1
    On wifi#connected do RuleTimer1 5 endon
    On rules#timer=1 do backlog IPAddress 1 endon
    On ipaddress1#data do websend [192.10.1.2] /?device=a&ipaddress=%value% endon
```
That should work or be close (not tested)

**Comment by vodkawasserfall** [comment]
```
Rule2
    On wifi#connected do RuleTimer2 5 endon
    On rules#timer=2 do backlog IPAddress1 endon
    On ipaddress1#data websend [192.10.1.2] /?device=a&ipaddress=%value% endon

Rule2 on
```
is what i'm using now. 

thank you, exactly what i wanted it to do. `backlog` and `#data` isn't obvious (to me) tough ;)

**Reply by barbudor** [reply]
Backlog is used in that case when you need the output of the command executed in a rule to be processed by another rule. Otherwise the command is executed but it's output doesn't go in the rule engine

#data is required for outputs that have a single value
